scipy/cluster/hierarchy.py

```diff
@@ -2990,17 +2992,16 @@ def set_link_color_palette(palette):
     """
     if palette is None:
         # reset to its default
-        palette = ['g', 'r', 'c', 'm', 'y', 'k']
+        palette = _link_line_colors_default
     elif type(palette) not in (list, tuple):
         raise TypeError("palette must be a list or tuple")
     _ptypes = [isinstance(p, str) for p in palette]
 
     if False in _ptypes:
         raise TypeError("all palette list elements must be color strings")
 
-    for i in list(_link_line_colors):
-        _link_line_colors.remove(i)
-    _link_line_colors.extend(list(palette))
+    global _link_line_colors
```

Comment:
hmm, is this needed? if so, then it deserves a comment, because otherwise someone else will try and remove it again